eclass/acct-user.eclass

```diff
@@ -469,26 +398,48 @@ acct-user_pkg_preinst() {
 acct-user_pkg_postinst() {
 	debug-print-function ${FUNCNAME} "${@}"
 
+	if [[ -n ${_ACCT_USER_ADDED} ]]; then
+		# We just added the user; no need to update it
+		return
+	fi
+
 	if [[ ${EUID} -ne 0 ]]; then
 		einfo "Insufficient privileges to execute ${FUNCNAME[0]}"
-		return 0
+		return
 	fi
 
-	if [[ -n ${ACCT_USER_NO_MODIFY} && -n ${_ACCT_USER_ALREADY_EXISTS} ]]; then
-		eunlockuser "${ACCT_USER_NAME}"
-
+	if [[ -n ${ACCT_USER_NO_MODIFY} ]]; then
 		ewarn "User ${ACCT_USER_NAME} already exists; Not touching existing user"
 		ewarn "due to set ACCT_USER_NO_MODIFY."
-		return 0
+		return
+	fi
+
+	local groups=( ${_ACCT_USER_GROUPS} )
+	local aux_groups=${groups[*]:1}
+	local opts=(
+		--comment "${_ACCT_USER_COMMENT}"
+		--home "${_ACCT_USER_HOME}"
+		--shell "${_ACCT_USER_SHELL}"
+		--gid "${groups[0]}"
+		--groups "${aux_groups// /,}"
+	)
+
+	if eislocked "${ACCT_USER_NAME}"; then
+		opts+=( --expiredate "" --unlock )
+	fi
+
+	if [[ -n ${ROOT} ]]; then
+		opts+=( --prefix "${ROOT}" )
 	fi
 
-	# NB: eset* functions check current value
-	esethome "${ACCT_USER_NAME}" "${_ACCT_USER_HOME}"
-	esetshell "${ACCT_USER_NAME}" "${_ACCT_USER_SHELL}"
-	esetgroups "${ACCT_USER_NAME}" "${_ACCT_USER_GROUPS// /,}"
-	# comment field can not contain colons
-	esetcomment "${ACCT_USER_NAME}" "${DESCRIPTION//[:,=]/;}"
-	eunlockuser "${ACCT_USER_NAME}"
+	elog "Updating user ${ACCT_USER_NAME}"
+	if ! usermod "${opts[@]}" "${ACCT_USER_NAME}" 2>"${T}/usermod-error.log"; then
+		# usermod outputs a warning if unlocking the account would result in an
+		# empty password. Hide stderr in a text file and display it if usermod
+		# fails.
+		cat "${T}/usermod-error.log" >&2
+		die
```

Comment:
I suppose some message here would be good given that this line will be without any context.